nixos/modules/services/web-apps/lubelogger.nix

```diff
@@ -0,0 +1,94 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.lubelogger;
+in
+{
+  options = {
+    services.lubelogger = {
+      enable = mkEnableOption "Lubelogger, a self-hosted, open-source, web-based vehicle maintenance and fuel milage tracker";
+
+      package = mkPackageOption pkgs "lubelogger" { };
+
+      dataDir = mkOption {
+        description = "Path to Lubelogger config and metadata inside of /var/lib.";
+        default = "lubelogger";
+        type = types.str;
+      };
+
+      port = mkOption {
+        description = "The TCP port Lubelogger will listen on.";
+        default = 5000;
+        readOnly = true; # Lubelogger does not allow you to configure the port it runs on
+        type = types.port;
+      };
+
+      user = mkOption {
+        description = "User account under which Lubelogger runs.";
+        default = "audiobookshelf";
+        type = types.str;
+      };
+
+      group = mkOption {
+        description = "Group under which Lubelogger runs.";
+        default = "lubelogger";
+        type = types.str;
+      };
+
+      openFirewall = mkOption {
+        description = "Open ports in the firewall for the Lubelogger web interface.";
+        default = false;
+        type = types.bool;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.lubelogger = {
+      description = "Lubeloger is a self-hosted vehicle maintenance and fuel mileage tracker";
+
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      environment.DOTNET_CONTENTROOT = "/var/lib/${cfg.dataDir}";
+
+      serviceConfig = {
```

Comment:
I imagine we'd want more hardening here as well, though I don't use the software myself atm